### ISSUE TITLE
Example of how to combine FmtSubscriber and FileAppender

### ISSUE THREAD

**Issue by ufoscout** [tokio-rs/tracing#1091]
Hi,
I am struggling to find a way to log to more than one output; specifically, I need to log to both stdout and a file.
Here is my attempt so far, but I can't even make it to compile:
```rust
        let env_filter = EnvFilter::from_str("info,my_package=debug").unwrap();

        let file_appender = tracing_appender::rolling::hourly("../target", "prefix.log");
        let (non_blocking_file_appender, guard) = tracing_appender::non_blocking(file_appender);

        let fmt_layer = fmt::layer();
        
        let subscriber = Registry::default()
            .with(env_filter)
            .with(fmt_layer)
            .with(non_blocking_file_appender);

        tracing_log::LogTracer::init().unwrap();

        set_global_default(subscriber).unwrap();
```

I would like that the env_filter applies to all outputs. I can't find a working example or documentation section about that, could you please suggest me a viable approach?

**Comment by davidbarsky** [comment]
This should work for you:
```rust
fn main() {
    let file_appender = tracing_appender::rolling::hourly("/some/directory", "prefix.log");
    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
    let filter = EnvFilter::from_str("info,my_package=debug").expect("Unable to create EnvFilter");

    let subscriber = tracing_subscriber::registry()
        .with(filter)
        .with(FmtLayer::new().with_writer(io::stdout))
        .with(FmtLayer::new().with_writer(non_blocking));
    set_global_default(subscriber).expect("Unable to set a global subscriber");
}
```

I believe that you can skip setting the `LogTracer`—the `FmtLayer` should initialize it for you if have the `log` feature enabled.

**Comment by ufoscout** [comment]
@davidbarsky 
I could not found any `FmtLayer`, I guess it is `fmt::Layer` from `tracing_subscriber`.
I got this working code now:
```rust
        let file_appender = tracing_appender::rolling::hourly("../target", "prefix.log");
        let (non_blocking, guard) = tracing_appender::non_blocking(file_appender);

let filter = EnvFilter::from_str("info,my_package=debug").expect("Unable to create EnvFilter");
        let subscriber = tracing_subscriber::registry()
            .with(env_filter)
            .with(Layer::new())
            .with(Layer::new().with_writer(non_blocking));

        tracing_log::LogTracer::init().map_err(|err| LoggerError::LoggerConfigurationError {
            message: format!("Cannot start the logger LogTracer. err: {}", err),
        })?;
        set_global_default(subscriber).map_err(|err| LoggerError::LoggerConfigurationError {
            message: format!("Cannot start the logger. err: {}", err),
        })?;
```

**Comment by ufoscout** [comment]
@davidbarsky 
I found the problem, I need to disable the ansi output:
```rust
Layer::new().with_ansi(false).with_writer(non_blocking)
```

I'll submit a PR with a small example for other's reference if you think it could be useful

**Comment by davidbarsky** [comment]
@ufoscout Glad you were able to figure your issue out!

> I'll submit a PR with a small example for other's reference if you think it could be useful.

I thought I had responded to this, sorry about that. I've added an example for this in #1106, but note that the example is using the new terminology described in https://github.com/tokio-rs/tracing/issues/1076, where tracing_core's `Subscriber` has been renamed `Collect` and tracing_subscriber's `Layer` has been renamed `Subscribe`.
